sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -210,7 +210,9 @@ abstract class StreamExecution(
     this, s"spark.streaming.${Option(name).getOrElse(id)}")
 
   /** Isolated spark session to run the batches with. */
-  private val sparkSessionForStream = sparkSession.cloneSession()
+  private val sparkSessionForStream = {
+    sparkSession.cloneSession(reuseArtifactManager = true)
```

Comment:
> Reusing the artifact manager will break resource isolation between
> parent and child sessions, making artifacts visible to both.
> Use this option only in well-reasoned scenarios.

Can we explain the "well-reasoned" scenario here? Should also fix the comment above.

Reply:
Done!
The cloned session will be used internally only for batch actions, so the user couldn't modify its artifacts.

Reply:
This feels a bit like premature optimization. Is there a strong reason (e.g. performance) for this?

Reply:
Yeah, performance is the reason. When there's a huge number of artifacts the clone might be slow.

Reply:
Let's get rid of this optimization to simplify the logic. If we see performance problems in streaming, we could rethink this again.